internal/api/v20240610preview/hcpopenshiftclusters_methods_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v20240610preview
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+
+	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/randfill"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+)
+
+func TestRoundTripInternalExternalInternal(t *testing.T) {
+	seed := rand.Int63()
```

Comment:
Can you make failing tests print their seed and provide an ergonomic way to re-run given a seed, so failures can be squashed and validated? Not sure if it's simple to do, but if you're able to wire this in to Go's built-in fuzzer, you'll get some of this for free.

Reply:
I wasn't able to wire up go's built in fuzzer for the same purpose, I'd end up rebuilding Daniel's impl.  I'll find a way to print the seed as a followup.